src/charm.py

```diff
@@ -123,19 +123,16 @@ def _unit_ip(self) -> str:
 
     def _on_config_changed(self, _):
         # For now, ignore the configuration
-        svc_was_running = False
         svc = SysbenchService()
         if svc.is_running():
             # Nothing to do, there was no setup yet
-            svc_was_running = True
             svc.stop()
-        if not (options := self.database.get_execution_options()):
-            # Nothing to do, we can abandon this event and wait for the next changes
-            return
-        svc.render_service_file(
-            self.database.script(), self.database.chosen_db_type(), options, labels=self.labels
-        )
-        if svc_was_running:
+            if not (options := self.database.get_execution_options()):
+                # Nothing to do, we can abandon this event and wait for the next changes
+                return
+            svc.render_service_file(
+                self.database.script(), self.database.chosen_db_type(), options, labels=self.labels
+            )
```

Comment:
If the service file is rendered before the prepare action is run, we'll trigger the `is_stopped` check in charm.py:75